managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -365,6 +372,50 @@ public void operationFailed(MetaStoreException e) {
         return updateCursorPropertiesResult;
     }
 
+    @Override
+    public CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties) {
+        MutableObject<CompletableFuture<Void>> updateCursorPropertiesResultRef = new MutableObject<>();
+        CURSOR_PROPERTIES_UPDATER.updateAndGet(this, map -> {
+            Map<String, String> newProperties =
+                    cursorProperties == null ? new TreeMap<>() : new TreeMap<>(cursorProperties);
+            if (map != null) {
+                map.forEach((k, v) -> {
+                    if (((String) k).startsWith(CURSOR_INTERNAL_PROPERTY_PREFIX)) {
+                        newProperties.put((String) k, (String) v);
+                    }
+                });
+
+                updateCursorPropertiesResultRef.setValue(asyncUpdateCursorProperties(newProperties));
```

Comment:
the new properties must take effect only after they are persisted to storage.
otherwise:
- in case of failure the value is no consistent on storage
- other readers of the value won't see the value consistently (who reads from this node may see something that will never been read)